docs/testbed/README.testbed.Testbed_v2.hld.md

```diff
@@ -0,0 +1,127 @@
+key                              = SWITCH_TABLE:switch_name
+; field                          = value
+HwSku                            = "FanoutRoot"/"FanoutLeaf"/"FanoutLeafSonic"
+ManagementIp                     =
+Type                             = "leaf_fanout"/"root_fanout"
+```
+* `SERVER_TABLE`
+```
+; Defines server medadata
+key                              = SERVER_TABLE:server_name
+; field                          = value
+HwSku                            = "Server"
+ManagementIp                     =
+ServerStatus                     = "active"/"down"
+```
+* `DUT_TABLE`
+```
+; Defines DUT metadata
+key                              = DUT_TABLE:dut_name
+; field                          = value
+HwSku                            = "DevSonic"
+ManagementIp                     =
+SonicHwSku                       =
+ProvisionStatus                  = "not_provisioned"/"in_progress"/"provisioned"
+```
+* `PORT_LIST`
+```
+; List contains physical ports of either a root/leaf fanout switch, DUT or server 
+key                              = PORT_LIST:<switch_name|dut_name|server_name>
+```
+* `PORT_TABLE`
+```
+; Defines port metadata
+key                              = PORT_TABLE:<switch_name|dut_name|server_name>:port_name
+; field                          = value
+BandWidth                        =
+VlanType                         = "access"/"trunk"
+PhyPeerPort                      =                                          ; physical peer port
+```
+* `VLAN_LIST`
+```
+; List contains VLAN ids assigned to a physical port
+key                              = VLAN_LIST:<switch_name|server_name>:port_name
+```
+* `VLANIDPOOL_SET`
+```
+; Set contains unique available VLAN ids
+key                              = VLANIDPOOL_SET
+```
+* `VIRTLINK_TABLE`
+```
+; a virtual link between DUTs
+key                              = VIRTLINK_TABLE:endport0:endport1         ; endport0 and endport1 are FK to `PORT_TABLE
+; field                          = value
+Status                           = "active"/"inactive"
+```
+
+## Implementation Plan
+* The whole process is divided into three stages:
+   1. stage#1: initial `connection_db` setup and provision
+   2. stage#2: dynamic vlan assignment support
+   3. stage#3: link state propagation support
+
+### Stage#1
+* In stage#1, only initial `connection_db` setup and provision is covered.
+* In the db provision during `add_topo`, the md5sum value of the connection graph file of current inventory will be calculated and check if it is in `LAB_CONNECTION_GRAPH_VERSIONS`.
+  * If yes, `add_topo` will skip provision the database because it indicates that current connection graph file had been used to provision the database.
+  * If no, `add_topo` will try to provision the database with the connection graph and add the md5sum value to `LAB_CONNECTION_GRAPH_VERSIONS` with current timestamp as score. Also, it will also trim `LAB_CONNECTION_GRAPH_VERSION` to only keep most-recent 20 entries.
```

Comment:
This part still seems a little scary to me. I don't know if any user should be able to make changes to the connection graph file and deploy those changes to the whole lab with an add-topo. Could we maybe check if their connection graph is up to date and prompt them to do a rebase if there is a diff?

Reply:
Only if a user wishes to provision the db with `enable_connection_db=True`, this check will happen. And if a user's connection graph file is dated, this `add_topo` will be aborted with an error prompting a rebase like you said.

Reply:
Right, but why do we give the user an option to provision the db?

Reply:
Yes, there is an option `provision_connection_db` added now.

Reply:
But _why_ does the user need an option to re-provision the db. I think this is related to my question above - is the user updating the physical layout of the lab or are they updating the logical connections/testbeds?

Reply:
Only the physical layout of the lab will be provisioned by user.

Reply:
Can you add a diagram/flow describing what the expected behavior in phase 1 is?

It's still not clear to me from the requirements or schema what is going on here. I think it would be helpful to me and others reviewing this to see how the testbed v2 is supposed to work when all of this is setup and provisioned.